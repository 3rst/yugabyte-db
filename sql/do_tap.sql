\unset ECHO
\i test_setup.sql
SET client_min_messages = notice;

-- $Id$

SELECT plan(26);
--SELECT * FROM no_plan();

CREATE OR REPLACE FUNCTION public.test_this() RETURNS SETOF TEXT AS $$
    SELECT pass('simple pass') AS foo
    UNION SELECT pass('another simple pass')
    ORDER BY foo ASC;
$$ LANGUAGE SQL;

CREATE OR REPLACE FUNCTION public.testplpgsql() RETURNS SETOF TEXT AS $$
BEGIN
    RETURN NEXT pass( 'plpgsql simple' );
    RETURN NEXT pass( 'plpgsql simple 2' );
    RETURN;
END;
$$ LANGUAGE plpgsql;

CREATE OR REPLACE FUNCTION public."test ident"() RETURNS SETOF TEXT AS $$
BEGIN
    RETURN NEXT pass( 'ident' );
    RETURN NEXT pass( 'ident 2' );
    RETURN;
END;
$$ LANGUAGE plpgsql;

SELECT is(
    findfuncs('public', '^test'),
    ARRAY[ 'public."test ident"', 'public.test_this', 'public.testplpgsql' ],
    'findfuncs(public, ^test) should work'
);

SELECT is(
    findfuncs('^test'),
    ARRAY[ 'public."test ident"', 'public.test_this', 'public.testplpgsql' ],
    'findfuncs(^test) should work'
);

SELECT * FROM do_tap('public', '^test');
SELECT * FROM do_tap('public'::name);
SELECT * FROM do_tap('^test');
SELECT * FROM do_tap();

/****************************************************************************/
-- Finish the tests and clean up.
SELECT * FROM finish();
ROLLBACK;
