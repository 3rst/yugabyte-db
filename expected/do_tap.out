\unset ECHO
1..26
ok 1 - findfuncs(public, ^test) should work
ok 2 - findfuncs(^test) should work
# public."test ident"()
ok 3 - ident
ok 4 - ident 2
# public.test_this()
ok 5 - simple pass
ok 6 - another simple pass
# public.testplpgsql()
ok 7 - plpgsql simple
ok 8 - plpgsql simple 2
# public."test ident"()
ok 9 - ident
ok 10 - ident 2
# public.test_this()
ok 11 - simple pass
ok 12 - another simple pass
# public.testplpgsql()
ok 13 - plpgsql simple
ok 14 - plpgsql simple 2
# public."test ident"()
ok 15 - ident
ok 16 - ident 2
# public.test_this()
ok 17 - simple pass
ok 18 - another simple pass
# public.testplpgsql()
ok 19 - plpgsql simple
ok 20 - plpgsql simple 2
# public."test ident"()
ok 21 - ident
ok 22 - ident 2
# public.test_this()
ok 23 - simple pass
ok 24 - another simple pass
# public.testplpgsql()
ok 25 - plpgsql simple
ok 26 - plpgsql simple 2
